test/e2e/storage_network_policy.go

```diff
@@ -0,0 +1,307 @@
+				Platform:       "openstack",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "manila-csi-driver-operator",
+				Platform:       "openstack",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      ManilaCSINamespace,
+				Name:           "openstack-manila-csi-controllerplugin",
+				Platform:       "openstack",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "smb-csi-driver-operator",
+				Platform:       "all",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "smb-csi-driver-controller",
+				Platform:       "all",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+		}
+		runDeploymentChecks(kubeClient, CSIdeploymentsToCheck, currentPlatform)
+	})
+
+	g.It("should verify required pod-selector for networkpolicy in openshift-cluster-storage-operator namespace", func() {
+		npList, err := kubeClient.NetworkingV1().NetworkPolicies(CSONamespace).List(context.TODO(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		checkNetworkPolicyPodSelectors(npList.Items, NPCSORequiredLabels)
+
+	})
+
+	g.It("should verify required pod-selector for networkpolicy in openshift-cluster-csi-drivers namespace", func() {
+		npList, err := kubeClient.NetworkingV1().NetworkPolicies(CSINamespace).List(context.TODO(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		checkNetworkPolicyPodSelectors(npList.Items, NPCSIRequiredLabels)
+
+	})
+})
+
+func runDeploymentChecks(clientset *kubernetes.Clientset, deployments []DeploymentCheck, currentPlatform string) {
+	results := []string{}
+	hasFail := false
+
+	for _, dep := range deployments {
+		if dep.Platform != "" && dep.Platform != currentPlatform && dep.Platform != "all" {
+			results = append(results, fmt.Sprintf("[SKIP] %s/%s (platform mismatch: %s)", dep.Namespace, dep.Name, dep.Platform))
+			continue
+		}
+
+		deployment, err := clientset.AppsV1().Deployments(dep.Namespace).Get(context.TODO(), dep.Name, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				results = append(results, fmt.Sprintf("[SKIP] %s/%s not found", dep.Namespace, dep.Name))
+				continue
+			}
+			g.Fail(fmt.Sprintf("Error fetching deployment %s/%s: %v", dep.Namespace, dep.Name, err))
+		}
+
+		missingLabels := []string{}
+		for key, val := range dep.RequiredLabels {
+			if deployment.Spec.Template.Labels[key] != val {
+				missingLabels = append(missingLabels, fmt.Sprintf("%s=%s", key, val))
+			}
+		}
+
+		if len(missingLabels) > 0 {
+			results = append(results, fmt.Sprintf("[FAIL] %s/%s missing labels: %s", dep.Namespace, dep.Name, strings.Join(missingLabels, ", ")))
+			hasFail = true
+		} else {
+			results = append(results, fmt.Sprintf("[PASS] %s/%s", dep.Namespace, dep.Name))
+		}
+	}
+
+	if hasFail {
+		summary := strings.Join(results, "\n")
```

Comment:
How about only print the `[FAIL]` items?

Reply:
Hmm, I don’t think it will generate too many logs. I just tried it and checked what was passed/skipped/fail in such a failure scenario.